yt_dlp/extractor/ninecninemedia.py

```diff
@@ -99,3 +100,40 @@ def _real_extract(self, url):
             }
 
         return info
+
+
+class CPTwentyFourIE(InfoExtractor):
+    IE_NAME = 'cp24'
+    _GEO_COUNTRIES = ['CA']
+    _VALID_URL = r'https?://(?:www\.)?cp24\.com/news/.*'
```

Comment:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?cp24\.com/news/(?P<id>[^?#]+)'
```
and use this id for the webpage download message
and maybe also return as `display_id`

Reply:
Looks ugly tbh but okay